### ISSUE TITLE
Documentation: Is "view" a reliable source for "parameters"

### ISSUE THREAD

**Issue by simonkowallik** [F5Networks/f5-appsvcs-templates#127]
### Is your feature request related to a problem? Please describe.

Request for clarification.

When creating an application via the FAST API the `"parameters"` JSON object contains the data for the template to render the AS3 declaration.

Example:
```json
{
    "name": "examples/simple_http",
    "parameters": {
        "tenant_name": "tenant1",
        "application_name": "appZ",
        "virtual_port": 1000,
        "virtual_address": "192.168.0.101",
        "server_port": 1000,
        "server_addresses": [
            "192.168.1.1"
        ]
    }
}
```

The response contains `"parameters"` as well:
```json
{
    "code": 202,
    "requestId": 29,
    "message": [
        {
            "id": "1796e028-7dd3-48e6-ae9f-03f179a79369",
            "name": "examples/simple_http",
            "parameters": {
                "tenant_name": "tenant1",
                "application_name": "appZ",
                "virtual_port": 1000,
                "virtual_address": "192.168.0.101",
                "server_port": 1000,
                "server_addresses": [
                    "192.168.1.1"
                ]
            }
        }
    ],
    "_links": {
        "self": "/mgmt/shared/fast/applications",
        "task": "/mgmt/shared/fast/tasks/1796e028-7dd3-48e6-ae9f-03f179a79369"
    }
}
```

When querying the specific Application, the response contains the `'view"` object within the constants in the rendered AS3 delcaration:

```json
{
    "class": "Application",
    "template": "http",
    "serviceMain": {
        "class": "Service_HTTP",
        "virtualPort": 1000,
        "virtualAddresses": [
            "192.168.0.101"
        ],
        "pool": "appZ_web_pool"
    },
    "appZ_web_pool": {
        "class": "Pool",
        "monitors": [
            "http"
        ],
        "members": [
            {
                "servicePort": 1000,
                "serverAddresses": [
                    "192.168.1.1"
                ]
            }
        ]
    },
    "constants": {
        "class": "Constants",
        "fast": {
            "template": "examples/simple_http",
            "setHash": "b4e574f5692532a03deb40b78599a7913ff63b2a31897a7b0168d99246d64715",
            "view": {
                "tenant_name": "tenant1",
                "application_name": "appZ",
                "virtual_port": 1000,
                "virtual_address": "192.168.0.101",
                "server_port": 1000,
                "server_addresses": [
                    "192.168.1.1"
                ]
            },
            "templateType": "local",
            "lastModified": "2023-03-23T15:50:57.427Z",
            "ipamAddrs": {}
        }
    },
    "_links": {
        "self": "/mgmt/shared/fast/applications/tenant1/appZ"
    }
}
```

Querying all applications the response is more concise but contains the same `"view"`:

```json
    {
        "template": "examples/simple_http",
        "setHash": "b4e574f5692532a03deb40b78599a7913ff63b2a31897a7b0168d99246d64715",
        "view": {
            "tenant_name": "tenant1",
            "application_name": "appZ",
            "virtual_port": 1000,
            "virtual_address": "192.168.0.101",
            "server_port": 1000,
            "server_addresses": [
                "192.168.1.1"
            ]
        },
        "templateType": "local",
        "lastModified": "2023-03-23T15:50:57.427Z",
        "ipamAddrs": {},
        "tenant": "tenant1",
        "name": "appZ",
        "_links": {
            "self": "/mgmt/shared/fast/applications/tenant1/appZ"
        }
    }
```

### Describe the solution you'd like

Is the returned `"view"` a safe source for the `"parameters"` submitted to FAST during creation/modification of the application?

Please document and clarify.

**Comment by simonkowallik** [comment]
Hi @shyawnkarim yes, I wonder if a) view is accurately representing the parameters submitted in the API request and b) whether these get updated when modified through the WebUI FAST interface.

**Comment by joelkeener** [comment]
Hello Simon, 
The OpenAPI spec's definition for the **view** in response to a GET of all applications or a specific one, is the same object that defines the required **parameters** object in a POST/PUT/PATCH to an application: ParametersDefinition.

I deployed an application, got all applications and the specific application to confirm the view matched the parameters I submitted, then updated the application with the GUI and confirmed the view was updated with the new parameters after my update.

**Comment by simonkowallik** [comment]
Great, thank you for the clarification!

It would be good if we can update the documentation with this information. The current [API reference](https://clouddocs.f5.com/products/extensions/f5-appsvcs-templates/latest/_static/apidocs.html#tag/Application-Management/operation/getFASTApplication) does not reflect that information:

<img width="624" alt="image" src="https://user-images.githubusercontent.com/1498073/234787483-48a551a8-34b9-4d0e-8b80-a34595aeea66.png">

**Comment by shyawnkarim** [comment]
This documentation update is being tracked internally with ID EC-217.

**Comment by shyawnkarim** [comment]
Closing. This issue was resolved with [Release 1.25.0](https://github.com/F5Networks/f5-appsvcs-templates/releases/tag/v1.25.0).
